resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -179,22 +131,85 @@ public DnsNameResolver(
+     * @param maxPayloadSize the capacity of the datagram packet buffer
+     * @param optResourceEnabled if automatic inclusion of a optional records is enabled
+     * @param hostsFileEntriesResolver the {@link HostsFileEntriesResolver} used to check for local aliases
      */
     public DnsNameResolver(
-            EventLoop eventLoop, ChannelFactory<? extends DatagramChannel> channelFactory,
-            InetSocketAddress localAddress, DnsServerAddresses nameServerAddresses) {
+            EventLoop eventLoop,
+            ChannelFactory<? extends DatagramChannel> channelFactory,
+            InetSocketAddress localAddress,
+            DnsServerAddresses nameServerAddresses,
+            int minTtl,
+            int maxTtl,
+            int negativeTtl,
+            long queryTimeoutMillis,
+            InternetProtocolFamily[] resolvedAddressTypes,
+            boolean recursionDesired,
+            int maxQueriesPerResolve,
+            boolean traceEnabled,
+            int maxPayloadSize,
+            boolean optResourceEnabled,
+            HostsFileEntriesResolver hostsFileEntriesResolver) {
 
         super(eventLoop);
-
         checkNotNull(channelFactory, "channelFactory");
-        checkNotNull(nameServerAddresses, "nameServerAddresses");
         checkNotNull(localAddress, "localAddress");
+        this.nameServerAddresses = checkNotNull(nameServerAddresses, "nameServerAddresses");
+        this.minTtl = checkPositiveOrZero(minTtl, "minTtl");
+        this.maxTtl = checkPositiveOrZero(maxTtl, "maxTtl");
+        if (minTtl > maxTtl) {
+            throw new IllegalArgumentException(
+                    "minTtl: " + minTtl + ", maxTtl: " + maxTtl + " (expected: 0 <= minTtl <= maxTtl)");
+        }
+        this.negativeTtl = checkPositiveOrZero(negativeTtl, "negativeTtl");
+        this.queryTimeoutMillis = checkPositive(queryTimeoutMillis, "queryTimeoutMillis");
+        this.resolvedAddressTypes = checkNonEmpty(resolvedAddressTypes, "resolvedAddressTypes");
+        this.recursionDesired = recursionDesired;
+        this.maxQueriesPerResolve = checkPositive(maxQueriesPerResolve, "maxQueriesPerResolve");
+        this.traceEnabled = traceEnabled;
+        this.maxPayloadSize = checkPositive(maxPayloadSize, "maxPayloadSize");
+        this.optResourceEnabled = optResourceEnabled;
+        this.hostsFileEntriesResolver = checkNotNull(hostsFileEntriesResolver, "hostsFileEntriesResolver");
 
-        this.nameServerAddresses = nameServerAddresses;
         bindFuture = newChannel(channelFactory, localAddress);
         ch = (DatagramChannel) bindFuture.channel();
+        ch.config().setRecvByteBufAllocator(new FixedRecvByteBufAllocator(maxPayloadSize));
+    }
+
+    private int checkPositive(int i, String name) {
+        if (i <= 0) {
+            throw new IllegalArgumentException(name + ": " + i + " (expected: > 0)");
+        }
+        return i;
+    }
+
+    private long checkPositive(long i, String name) {
```

Comment:
As I said in a previous comment, I think there would be some value in extracting those into a `Checks` static class. My personal choice would be to rename `ObjectUtil` that currently only contains `checkNotNull`.